### DISCUSSION TITLE
Add an option (manily for Android) to keep Picker open after creating filters

### DISCUSSION THREAD

**Discussion by u-RraaLL** [uBlockOrigin/uBlock-issues#1929] [Ideas]
This [post](https://www.reddit.com/r/uBlockOrigin/comments/s2ucan/multiple_selections_on_android/) made me wonder if it wouldn't be a good idea if there was _an option_ to let Picker stay open on Android after tapping [Create].

The reasoning: If a user wants to hide multiple items, they'll have to tap the 3-dot menu, then Addons, then uBO, then Picker for every single item (that's four unnecessary steps). That's gonna get tiresome pretty quick. And the copy/paste methods I described in the reddit post aren't that much better.

Personally, if I want to hide some mobile items, I'll usually just use FF's RDM on my PC and then import the filters to my phone as that seems less of a hassle. But that's still not as convenient as "picking" multiple items at once or, in this case, having the Picker stay open for easy "second picking" ;)

What do you think?

**Comment by gorhill** [comment]
The poster on Reddit does not provide any concrete use case for us to find out on our own, I typically reject all such requests to add code or modify the behavior of existing code from just someone asking for it without making the case with concrete, real world examples of usefulness. I need to see the cases for myself, this is the only way to convince and motivate to throw code at uBO.

**Reply by u-RraaLL** [reply]
Like I said, I usually use desktop to clean up mobile sites to my preference and import the filters to my phone, but if I'm somewhere without access to a PC, I'll usually have to use picker multiple times.

If you require examples, I'll try to send some over the weekend.
Usually it's weather sites as they're often full of annoying widgets I don't need. Sometimes shops or webnovel sites.

**Reply by u-RraaLL** [reply]
Let's take `weather.com` as an example. Location: Berlin.

Below are screenshots with DOM Inspector on to show sections that are unnecessary to me.

<details><summary>Screenshots</summary>

![Screenshot_20220116-033721](https://user-images.githubusercontent.com/92610420/149646201-b1ceed70-38f4-487d-89f1-5572c582619e.png)
![Screenshot_20220116-033804](https://user-images.githubusercontent.com/92610420/149646205-68e3095e-f9de-435e-867d-fc1f80785e87.png)
![Screenshot_20220116-033828](https://user-images.githubusercontent.com/92610420/149646208-5de61a9a-2b3f-4cc0-9d59-babf3adfae0e.png)
![Screenshot_20220116-033847](https://user-images.githubusercontent.com/92610420/149646213-d320e0e8-6629-4149-9818-acef4c4b2a0f.png)

</details>

Counting the button, that's 7 times I had to use the Picker. That means 6 "unnecessary" times I had to go to `the 3 dot menu > Add-ons > uBO > Picker`.

Only the last section is in the annoyance lists (fanboy's).

<details><summary>Picker-crafted filters</summary>

```
weather.com###WxuContentMedia-main-1bf79a62-0cd5-41f8-a941-3906d3b9355e
weather.com###WxuTodayMapCard-main-61bb1389-287a-4e88-9e75-5f36f3179775
weather.com###WxuContentMedia-main-154378c9-48db-470b-a957-20aa6e9d84b8
weather.com###WxuContentMedia-main-871dfd0b-cb01-4671-8807-cf48de2fdf4e
weather.com###WxuUpsellCard-main-5e7c539c-32b6-497d-9a5b-14e495198e3c
weather.com###WxuPromoDriver-main-721f1a1a-460a-422c-a0db-db83a582b227
weather.com###WxuContentMedia-main-154205c3-43e0-4e4d-a284-972095b12084
weather.com##.Navigation--severe--2oLia.Navigation--navigationLink--3hZUC
```
</details>

There's actually 8 filters. One more is for the 10day tab, which has featured deals section (also covered by FA).

There's no easy way for an average Joe to hide multiple items per page on Android. Letting Picker stay open would make that much easier.

I even tried using the desktop's window version of the popup to see if that'd work (`popup-fenix.html`), but it's not looking like it's compatible with Android. I was somehow able to make the domain appear for the first tab (reddit), but the power button stayed white, zapper, picker and report buttons were gone, AND after trying the cosmetic switch, the refresh button auto hid, the crossed out button returned to normal (although the switch worked after refreshing reddit), which made it impossible to use the window-popup to disable the switch again.

If the "window-popup" actually worked on Android, and it wouldn't be necessary to manually change tab numbers in the address bar - instead, have a drop-down menu, like in the logger, that'd be pretty cool. You could leave the tab open and not have to use the 3 dot menu > Add-ons > uBO. At all.

But I imagine that'd take more work than letting Picker stay open.

If you need more example websites, I can provide them, but the idea is pretty much the same for any website `¯\_(ツ)_/¯`

Okay, I think I've made my case. Ultimately it's your free time you'd be sacrificing, so I'll respect whatever decision you're gonna make.

Enjoy your weekend.

**Reply by u-RraaLL** [reply]
Never mind, I forgot you can keep the popup open in the logger, which significantly lessens the number of taps necessary to multi-hide things on mobile.

**Comment by u-RraaLL** [comment]
https://github.com/uBlockOrigin/uBlock-issues/discussions/3057

I think the most user-friendly way to implement this would be to add a "pin" icon in/above one of the top corners of picker.
An icon which, if clicked, would keep picker open even after clicking the [Create] button.

**Reply by garry-ut99** [reply]
### 1

> garry-ut99 : Rather a duplicate of (...)
> > u-RraaLL : Not exactly. The issue was for desktop 

Ok, lets the truth stay somewhere between me and you, it'll be most fair, here is why : 
https://github.com/uBlockOrigin/uBlock-issues/issues/467 mentioned **Desktop** and https://github.com/uBlockOrigin/uBlock-issues/discussions/3057 mentioned **Desktop** and Mobile, 
there was an accent on Mobile, but it's not clear whether the purpose of the accent was :
a) only a distinction which platform is more cumbersome, and he still wanted to solve the issue for both platforms
b) or that his issue was aimed at Mobile platform only
The blame is on the OP who wasn't clear enought.

> u-RraaLL : and the original title was looking for a shortcut (while keeping picker open was just an alternative)

Maybe, but still the both options were listed as solutions : 
https://github.com/uBlockOrigin/uBlock-issues/issues/467 mentioned a shortcut or **keeping picker open** and https://github.com/uBlockOrigin/uBlock-issues/discussions/3057 mentioned **keeping picker open**

<br>

In the end the both issues mentioned **Desktop** and **keeping picker open**, but your reasoning is still reasonable, thus the truth is somewhere in the middle (50% me / 50% you), let his issue be a duplicate of whichever the two other issues anybody thinks.

<hr>

### 2

> > u-RraaLL : The main reason to keep picker open would be to improve the ease of use on mobile.

Which doesn't change a fact it could be improved on Desktop as well.
Again the truth is 50% me / 50% you.


>  garry-ut99 : then an alternative and the easiest & least cumbersome solution, would be to add an additional 2in1 `Create&Pick` button
> > u-RraaLL : On the contrary, this would be totally cumbersome.

No when taking into account what I mostly meant: it requires the least number of clicking & time than any other solutions, but I have to admit that I didn't notice that there might be no room for another button on Mobile, while there is a room on Desktop, however this doesn't change a fact that it's the simplest & easiest solution out here, and calling it "totally cumbersome" is exaggerated, if not to say that this is plain trolling.

> > u-RraaLL : There is no space for more buttons on mobile

Not exactly, this is only 50% true, this is not problem on Desktop, and not problem on Mobiles with standard screen (like Galaxy S) which ironically still have a place for one more button, (but yes, on bigger phones like Galaxy S+ we should not waste space) :

Galaxy S10 _______________________ GalaxyS20+
<img src="https://github.com/uBlockOrigin/uBlock-issues/assets/72945564/2cd11690-7dda-47c4-a03f-f6c95da0f92b" width="200">|<img src="https://github.com/uBlockOrigin/uBlock-issues/assets/72945564/b9da8d78-a837-41d9-8c4f-14e4d81955e5" width="210">
![S10]()|![S20]()

> u-RraaLL :  I think the most user-friendly way to implement this would be to add a "pin" icon in/above one of the top corners of picker. An icon which, if clicked, would keep picker open even after clicking the [Create] button.

Only 50% true again, another idea to consider, but not necessarily "the most user-friendly" 
(or using your words: "On the contrary, this would be totally cumbersome"), 
the reason is that after Picker window stays "always on top", an user might have troubles to pick elements if they are  obstructed by Picker window, and then an user will be forced to scroll website to unhide elements he wants to pick, because the visible screen area will be reduced.

<hr>

### 3

> u-RraaLL : Anyway, let's continue in the discussion I linked..

Ok, but it's strange, especially after you previoulsy started proposing ideas and answering comments in the previous discussion, which means you already did choose the previous place for a discussion, you could've stay there, instead of suddenly moving here to brute-force your point of view that it must be a duplicate of your issue.

**Reply by u-RraaLL** [reply]
> Which doesn't change a fact it could be improved on Desktop as well.

I agree. I'll rename the discussion.

> there is already not enought space for already existing buttons

Well, that is most certainly a bug. Too bad it wasn't reported before or even a bit earlier - maybe @gorhill would've fixed it before submitting 1.55.

The reason I said there's not enough space is because the empty space between [Preview] and [Create] is important. It is the drag area that allows you to move picker around.
I know I make use of this on both desktop and mobile. Having the area smaller or non-existent (like you show it already is on other langs) would make that impossible.

> always on top

I didn't say that. I said it'd let it stay open after hitting [Create]. Picking would work as usual.

> in the previous discussion

Well, I intended the discussion to close after commenting duplicate. I didn't consider discussions behave differently from issues and  clicking "Reply" wouldn't "Close as a duplicate". I only noticed it stayed open because you replied.
Of course I'd rather keep the discussion going in the 2 year prior open discussion rather than in a duplicate one.

**Reply by garry-ut99** [reply]
> > garry-ut99 : Which doesn't change a fact it could be improved on Desktop as well.
> 
> u-RraaLL : I agree. I'll rename the discussion.

Then it becomes a duplicate of https://github.com/uBlockOrigin/uBlock-issues/issues/467, should be locked and continued in `#467` (unless many comments and accent on Mobile is sufficient enought to make an exception and let it stay separately).

> u-RraaLL :  Well, I intended the discussion to close after commenting duplicate. 
> u-RraaLL :  I only noticed it stayed open because you replied. 

You proposed an idea (the logger-workaround) and this is what caused the discussion to keep going, and I just replied in the same place, like in the previous similiar case : `https://github.com/uBlockOrigin/uBlock-issues/issues/3050#issuecomment-1873039698` , which means it stayed open because you initially throwed an idea, not because I replied. I'm blamed two times already.

<hr>

> u-RraaLL :  (...) the drag area (...)

I tried to drag by drag-area (like on Screenshoot 1) but it didn't move, the screen was scrolling instead, it also doesn't have the drag area marked with a different texture, like on Desktop, like it was never possible to drag on Mobile, and as for Screenshoot 2 there is no a drag-area at all.

**Reply by u-RraaLL** [reply]
> I tried to drag by drag-area (like on Screenshoot 1) but it didn't move, the screen was scrolling instead, it also doesn't have the drag area marked with a different texture

A bug, like I said.

The area is visible and works fine on English or other "short" languages I tried.

It doesn't work/show when the buttons are wider. Which is not the case on desktop.

@gorhill Could you look into this?

**Reply by garry-ut99** [reply]
Ok, given you've provided more information, I can confirm now that definately this is a bug on some languages.

**Reply by garry-ut99** [reply]
The bug fixed now in uBO `1.56.1b10` in https://github.com/gorhill/uBlock/commit/953c978d59

**Reply by u-RraaLL** [reply]
Yes, noted already in the new pinned issue for informally reported issues: https://github.com/uBlockOrigin/uBlock-issues/issues/3151

**Reply by gorhill** [reply]
Now that there is a "window bar", I am thinking about replacing the Quit button with a close button icon in the window bar, and also adding a contract/expand button icon to minimize/expand dialog, and even maybe a pin button icon to prevent the dialog from fading away.

**Reply by garry-ut99** [reply]
Noted where, in the issue you linked (3151) is nothing mentioned about Picker.

<hr>

> gorhill : Now that there is a "window bar", I am thinking about (...)

Assuming that "window bar" means the Picker's new drag bar, yes - it's a good idea to introduce changes you've mentioned, some summarision : 

> 1) replacing the Quit button with a close button icon in the window bar

It would fix that nasty "doubled row of buttons" issue mentioned in https://github.com/uBlockOrigin/uBlock-issues/discussions/1929#discussioncomment-8006331 point 2 and visible on the left screenshoot, which wastes one row of precious space and occurs on standard and small sized smartphones, for example standard size lines:  Galaxy S main line (not `+` / `ultra`) and  Apple Iphone mainline (not `+` / `Max`) & Iphone SE. **EDIT:** By the way - I've noticed that it's already fixed for Galaxy S9, but for example not for S10.

**EDIT**: I've checked again and rethinked a bit, and I think it's better to keep Close button in the current place, and just replace "Close" text with "X" text or icon, such way that it would make the button narrowed and having less width - lets say at least 2x less width, it would fix the "doubled buttons row" issue on most standard size phones, and at the same time the button would stay in same place, where it's much easier to click it by thumb finger when keeping and using phone in one hand, because on the drag bar it's too far if using a hand on the opposite side of the button (for example when using left hand while the button is on the right side of drag bar, or vice-versa), unless on the drag bar, the Close button would be on the both sides, so that using phone in any hand would be easy to click the button.

 (_by the way sometimes it's required to click damn "show previous replies" button to make the link work and to jump to that comment, because it's too stupid to automatically unhide the comment on it's own, on the other side it has no problems with automatically unhiding comments in normal issue threads_)

> 2) and also adding a contract/expand button icon to minimize/expand dialog

It would add an alternative method to hide / show Picker window to the current method (tapping screen right/left)

> 3) and even maybe a pin button icon to prevent the dialog from fading away.

Would fix the current thread issue "to keep Picker open after creating filters " by implementing a solution proposed in this comment: https://github.com/uBlockOrigin/uBlock-issues/discussions/1929#discussioncomment-8003788

**Reply by u-RraaLL** [reply]
> Noted where

Currently 2nd from the bottom in the issues list.

---
> Now that there is a "window bar", I am thinking about replacing the Quit button with a close button icon in the window bar,

One thing I'm worried about with the "window bar" approach is how easy it's now going to be to miss-click the area above the picker, which resets the window and relaunches "Pick". So if you're trying to fine-tune a filter and want to drag picker around to see how well it matches as you scroll the page, a slip of the finger will reset your crafted filter.

> and even maybe a pin button icon to prevent the dialog from fading away.

Do you mean the autofade which happens on FF's RDM or by right-swiping on mobile?

Or precisely what I just talked about above? In which case, a button there would still be kinda risky if miss-clicked - unless clicking it would be the first thing you'd do OR the pin status would be remembered by uBO when re-launching picker.

One way of preventing miss-clicks would be to add some invisible padding above the window bar, another would be to disable the click-away gesture altogether, considering the existence of the "Pick" button.
And yes, I'm aware that the gesture also exists on desktop. And I must admit that I sometimes fall into its clutches when switching between windows or tabs. So I wouldn't mind a way of disabling that on desktop either.

Or did you mean it as garry understood it: implement it for the idea of keeping picker open after creating filters?

>  and also adding a contract/expand button icon to minimize/expand dialog

What do you have in mind here? Hide the filter suggestions to save on space when trying to fine-tune filters manually?

**Reply by garry-ut99** [reply]
> u-RraaLL : Currently 2nd from the bottom in the issues list.

Nothing about Picker, only this : 

> - [x] Add a tool to benchmark scriptlets lookup ([see](https://github.com/uBlockOrigin/uBlock-discussions/discussions/878))

Unless the whole pinned issue was moved to team discussions and updated, where I have no access.

**Reply by u-RraaLL** [reply]
Now it's the 2nd "solved" from the bottom. I added a new bottom item after my comment here because it reminded me about something else. But I also added a 2nd link (to here) in the solved item, because the original one linked only to an internal discussion.

> - [x] Address element picker issues on mobile ([see](https://github.com/uBlockOrigin/uBlock-discussions/discussions/871#discussioncomment-8497646), [earlier](https://github.com/uBlockOrigin/uBlock-issues/discussions/1929#discussioncomment-80118520))

Then again, considering gorhill's followup here and our replies, I wonder if I should untick that item for now?

**Reply by u-RraaLL** [reply]
@gorhill I'm not too fond of the rearranged named buttons. Particularly, the [Preview] button being in the dead center seems kinda weird. Why not stick it to the right?

When minimizing picker, why not keep the window bar in place and fold the window upwards?
Currently when picker is in the bottom half of the screen and/or the right side, the collapsed bar runs away from the mouse which isn't very convenient. The obstruction of the screen will be minimal and the behavior is basically already the same if the picker is in the upper left quadrant.

**Reply by gorhill** [reply]
> being in the dead center seems kinda weird.

Would you have found this weird if it has been like this since the beginning?

I put it in the middle because from the left to right the order is more logical: first you pick, then you preview, then you create. People will get use to it -- if I had chosen this since the beginning, nobody would question that it was like this, the issue is the change, not that somehow the order before was the result of some special wisdom -- it could also be argued against. I always try to keep in mind that people dislike change but not to the point of being stuck and unable to do anything.

> When minimizing picker, why not keep the window bar in place and fold the window upwards?

I chose to fold to the closest edges, this also solve the issue of having the minimized dialog moved around where it could easily end up outside the viewport when un-minimized. While I develop and test all this before committing I find these cases and I have to make choices to avoid such issues. I could assign a shortcut to toggle minimize.

**Reply by u-RraaLL** [reply]
> Would you have found this weird if it has been like this since the beginning?

Yes, I do wonder about that... But it's not exactly the order of buttons that bothers me, it's mainly the size of the gaps between them.
It's not an issue on mobile where the gaps are much smaller.

I've done some simulations and this looks a bit better to me: https://i.imgur.com/Sa87W8v.png
Likely because it's closer to the old layout, yes, but also because button sizes don't differ by that much and there's a bit more padding on the edges, which also closes the gaps by a tiny margin.
So even if the preview button stays in the center, it's the actual center this time: https://i.imgur.com/ZtOUcqO.png

Could you consider adding `padding-inline` for the toolbar (or whatever way of increasing padding by the left and right edges you prefer) and maybe assigning `min-width` to the buttons (at least on desktop)?

> where it could easily end up outside the viewport when un-minimized.

I see. I had not thought of that.

>  I could assign a shortcut to toggle minimize.

Yes, that would likely be helpful.

**Reply by garry-ut99** [reply]
I've noticed 3 major issues in Picker on Firefox Mobile:

with "Scroll to hide toolbar" enabled by default, as per https://github.com/uBlockOrigin/uBlock-issues/issues/3058#issuecomment-1877005118
- **1a)** the initial position of Picker window is too low, hence the bottom part of Picker window (3 rows of text, which is significant), is hidden which is an issue itself because each time Picker is opened it needs to be manually moved up to unhide the bottom part
- **1b)** another issue related to it is: that after clicking minimize `_` button, without unhiding the bottom part of Picker window, the minimized Picker window bar disappears completely and is hidden completely, hence cannot be recovered anymore, leaving Picker in a broken state, with the highlight of the highlighted element, the only way to unbreak Picker, is to swipe right to disable Picker, andthen invoke Picker again since the beginning

![4](https://github.com/uBlockOrigin/uBlock-issues/assets/72945564/4363a1cc-1eb9-4127-af8e-f47a3c6a0390)

(unrelated to "Scrool to hide tollbar") :
- **2)** in edit field of Picker, text sometimes has normal font size, and sometimes bigger font size (almost 2x), it's especially annoying when moving the left slider to right or to left, which results in constant font size switching, can be reproduced in the current thread: try to pick text in the first comment, and keep moving the left slider in both directions alternately

(for some reason the video plays fine only in Chromium-based browser, and fails to load on Firefox with a message:
`No video with supported format and MIME type found`)

https://github.com/uBlockOrigin/uBlock-issues/assets/72945564/34f93740-8c9a-4891-857c-2007836b42e7

All the above issues occur only on real mobile phones, and cannot be reproduced in Desktop or Responsive Design Mode: 
- **1a** + **1b** due to lack of  "Scroll to hide tollbar" feature in Desktop & Responsive Design Mode
- as for **2** I don't know why

~~I currently can't send any screenshoots or videos of issues, due to file transfer problem between my smartphone & PC.~~ EDITED: The problem is gone, I've uploaded the files.

**Reply by u-RraaLL** [reply]
Do the "scroll to hide toolbar" issues happen when the toolbar is on the bottom only? Or regardless its placement? Just curious - either way, it's something that'll need to be addressed.

**Reply by garry-ut99** [reply]
Regardless of its placement.

**Reply by gorhill** [reply]
How do you load dev build version in your phone?

**Reply by gorhill** [reply]
> Likely because it's closer to the old layout

I don't mind grouping the first two buttons together. The idea by having them evenly spaced was to make fat fingers less likely to press the wrong button on mobile.

> Could you consider adding padding-inline for the toolbar

Anything that consumes horizontal space--like adding horizontal padding--is going to make it more likely to have all the buttons not fit in a single line in some languages.

Also, it's quite subjective, I don't see why padding the toolbar to move the buttons away from the edges is a positive. For my taste, I like to have the buttons spaced from the edge like all the other visuals in the picker.

> maybe assigning `min-width` to the buttons

I will assign a minimum width, but still not guarantee that all buttons will have same width in other languages.

**Reply by garry-ut99** [reply]
> How do you load dev build version in your phone?

https://www.reddit.com/r/firefox/comments/18khd3i/firefox_nightly_for_android_now_allows_addon/
Currently it works even on normal Firefox Mobile release (not required Beta or Nightly anymore).

**Reply by gorhill** [reply]
> works even on normal Firefox Mobile release

I have Firefox 123 but I don't see the "Install add-on from file" entry.

---

I also do not see it in Nightly.

**Reply by u-RraaLL** [reply]
Read the beginning of the post. Go to about firefox, tap the logo 5 times to enable developer mode. At first, I missed it too.

---

> The idea by having them evenly spaced was to make fat fingers less likely to press the wrong button on mobile.

That's what I guessed, but there's more space now with "quit" moved up, so I don't think having two buttons grouped should cause issues.

> For my taste, I like to have the buttons spaced from the edge like all the other visuals in the picker.

I get that, but the buttons and everything else (e.g. the input area) are now much closer to the window edges. Sure it's only 2px on each side, but that amounts to half the previous size. And considering the above, we can now test if any language requires sacrificing these 4px.

**Reply by gorhill** [reply]
> tap the logo 5 times

I did this, I actually knew about this already since a long time as I have enabled debugging through USB in the past. I just do not have that "Install add-on from file" entry:

<details><summary>Screenshot</summary>

![Capture+_2024-03-03-11-00-57](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/3330f36c-6bab-40b5-8600-5323d552d136)

</details>

**Reply by gorhill** [reply]
<https://bugzilla.mozilla.org/show_bug.cgi?id=1868940>:

> Confirming that the option is still available and functionable on Android 10 or higher devices.

I am on Android 9.

**Reply by garry-ut99** [reply]
I am on 10/11.

Still you can try to downgrade temporarily to FF 122/121 or to install separately FF 122/121 and to rename `xpi` to `zip` maybe works.

**Reply by u-RraaLL** [reply]
I found the longest uBO language that is available in FF. It's Malayalam.

While it nearly fills the screen with the buttons, there's still enough space to bring back the old 4px padding. My phone isn't big by any means either.

Probably the 2nd longest language in terms of picker buttons is Bulgarian, but it's roughly 40% shorter. There should be no issues with fitting the buttons anymore.

<details><summary>screenshot of picker in Malayalam</summary>

![Screenshot_20240303-182528](https://github.com/uBlockOrigin/uBlock-issues/assets/92610420/0031c4d3-c486-44b2-9f66-311bee10b19d)
</details>

**Reply by gorhill** [reply]
> Sure it's only 2px on each side, but that amounts to half the previous size

Is `2px` causing any functional issue?

On my side, taking into account the [`min-width` now set on button](https://github.com/gorhill/uBlock/commit/43e0e15125b6993f23b02d76f4530a8c32ff56c4#diff-582d359db626444e5383bc21d135314bb833e6dfcbee5bb3635f4231f67dca08R49), `4px` causes stacking:

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/1206f26e-30be-4f17-b905-d8c30d094b81)

Sure, I can adjust the `min-width` _yet again_, but with the goal of freeing as much horizontal space as possible, I can't stop worrying about everybody sense of esthetic and aim for make stacking as unlikely as possible.

Here is the thing, if the arguments are just subjective sense of esthetic, I rather make my life simpler by going for safe and simple and make available as much horizontal space as possible, there is no guarantee that translations won't change in the future.

So for now I will consider convincing functional issues, not discuss sense of esthetic -- I can't understand why `2px` is an issue.

**Reply by garry-ut99** [reply]
Or maybe to wrap long text line in buttons into 2 rows :
https://stackoverflow.com/questions/7469088/html-can-i-display-button-text-in-multiple-lines

 <img src="https://github.com/uBlockOrigin/uBlock-issues/assets/72945564/cabccde8-02d8-427b-8db8-cfb9e935a3d5" alt="alt text" height="40">

**Reply by gorhill** [reply]
> Still you can try to downgrade temporarily to FF 122/121 or to install separately FF 122/121 and to rename xpi to zip maybe works.

I'm really out of luck -- I have a mobile phone which bootloader can't be unlocked, so upgrading to lineageos is not an option.

As for an older version of Firefox, I don't see this option offered by Firefox to install older versions.

---

From [here](https://ftp.mozilla.org/pub/fenix/releases/121.1.0/android/fenix-121.1.0-android-arm64-v8a/) I guess...

**Reply by krystian3w** [reply]
Or: https://github.com/mozilla-mobile/firefox-android/releases/tag/fenix-v121.1.0

**Reply by stephenhawk8054** [reply]
I can install FF 121 on another profile in android 10, but I don't see the option "Install add-on from file" either

**Reply by garry-ut99** [reply]
> gorhill :  As for an older version of Firefox, I don't see this option offered by Firefox to install older versions.

I meant to uninstall FF 123 for a while, and to install FF 122/121 temporarily instead, but let's forget about it, 

because I also mentioned about a better solution - about installing another FF separately: 
FF Beta/Nightly and FF Release can be installed separately at the same time, 
just like on Desktop, because they are treated as separate/different programs, 
I've just installed on my phone FF Beta 122.0b9 without uninstalling FF Release 123: 
`https://apkcombo.com/pl/firefox-beta/org.mozilla.firefox_beta/old-versions/122.0b9/`
or if you already use FF Beta/Nightly, then can install FF 122 Release: 
`https://apkcombo.com/firefox/org.mozilla.firefox/old-versions/122.0/`

But I'm not sure whether "Install add-on from file" will appear in Android 9 + FF Beta 122.0b9 / FF Release 122, 
if not then can try FF Nightly 122.0a1 or FF Beta 122.0b1:
`https://apkcombo.com/firefox-nightly-for-developers/org.mozilla.fenix/old-versions/122.0a1/`
`https://apkcombo.com/firefox-beta/org.mozilla.firefox_beta/old-versions/122.0b1/`

> stephenhawk8054 : I can install FF 121 on another profile in android 10, 

Yes, possible to install in a separate profile if one needs to install different versions of the same FF channel, 
but like mentioned above, different FF channels can coexists at the same time in same profile.

> garry-ut99 : to install FF 122/121 
> stephenhawk8054 : but I don't see the option "Install add-on from file" either


I've checked and it seems that "Install add-on from file" has been introduced in FF 122 and not in FF 121, 
I confirm it does exist in FF Beta 122.0.b9, I didn't check in FF Release 122 but I  guess it does exist as well.

**Reply by stephenhawk8054** [reply]
Ah yeah "Install add-on from file" is indeed in FF 122 stable

**Reply by ghost** [reply]
https://extensionworkshop.com/documentation/develop/getting-started-with-web-ext/#testing-in-firefox-for-android

**Reply by krystian3w** [reply]
Is too in 123 at Android 13.

Also apk combo should be last domain to download beta/nighty due we have apk at GitHub repo: https://github.com/mozilla-mobile/firefox-android/tags/

* https://github.com/mozilla-mobile/firefox-android/releases/tag/fenix-v121.0b9
* https://github.com/mozilla-mobile/firefox-android/releases/tag/fenix-v122.0b9

**Reply by gorhill** [reply]
Yes, I ended up following the [advice in bugzilla comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1880777#c4) to install web-ext, and I could run local copy of uBO within minutes, it all worked very well.

It's just silly me who was resisting installing an npm package in order to avoid pulling in who-knows-how-many dependencies with the risks that come with this, but in the end everything work very well. I didn't want to install with `--global` though, but that still worked fine, just need to create aliases to avoid having to enter long commands.

**Reply by garry-ut99** [reply]
> garry-ut99 : ( https://github.com/uBlockOrigin/uBlock-issues/discussions/1929#discussioncomment-8656975 ) : I currently can't send any screenshoots or videos of issues, due to file transfer problem between my smartphone & PC. 

The problem is gone, I've uploaded the files.

<hr>

Update 12-03-2024 : Issues 1a & 1b have been fixed for me now in 1.56.1b15. Issue 2 not yet.

**Comment by krystian3w** [comment]
As for me, an example would be a pop-up what occupies the entire screen area and has outer margins.


As a result, the picker will block only the pop-up the first time and only the background below it the second time.

Building pop-ups is popular in that way and the AdGuard Pop-ups and EasyList Pop-ups list won't all cut it in time or mark as useful pop-ups for rest community.
